### DISCUSSION TITLE
POST Add Project Member api issue

### DISCUSSION THREAD

**Discussion by ec-dev-1** [iTwin/itwinjs-core#3127] [Q&A] [answered]
Hello,

I have a question about the Add Project Team Member API
https://developer.bentley.com/apis/projects/operations/add-project-team-member/

According to the Docs, when creating the POST body, we need to include either a userID or an e-mail.
When I tested using an email (tried using "email" and "e-mail"), the response stated that a "User id was missing or could not be determined" (see below).

I formatted my request body based on the Request example on the Docs that uses email.

Can anyone tell me what I might be doing incorrectly?

Thank you

notes:  
- scope  projects: modify was added
- User creating the request have administration_invite_member permission

Request:
`
POST /projects/541f29c8-df68-4308-8beb-ca78d606c027/members HTTP/1.1
Host: api.bentley.com
Authorization: Bearer {{token}}
Content-Type: application/json
Content-Length: 70
{
    "email": "john.doe@email.com", "roleNames": ["Team Member"]
}
`

Response:
`
{
  "error": {
    "code": "InvalidTeamMemberRequest",
    "message": "Cannot add team member to project.",
    "details": [
      {
        "code": "MissingRequiredProperty",
        "message": "User id was missing or could not be determined",
        "target": "request body"
      }
    ]
  }
}
`

**Comment by davidhjones** [comment, marked as answer]
The API request looks syntactically correct to me, specifically `email` is correct.

The issue you are facing is from trying to invite a user which does not have a Bentley Account (in your example `john.doe@email.com`). The `email` (or `userId`) passed in must correspond to a user which has a valid Bentley Account. 

Additionally, it looks like you're attempting to assign the `Team Member` role. This role is implicitly assigned with all project initiations, so its redundant that you're explicitly assigning it. You typically want to assign an additional Project Role as part of this API request (or not assign any roles explicitly, in which the Team Member role will be implicitly assigned).

**Reply by ec-dev-1** [reply]
@davidhjones  thank you for your reply.  I was indeed adding a new team member who do not have a Bentley account.  After having this user sign up for one, I was able to successfully add this user.  Thank you once again.
